src/main/java/com/palantir/semver/SemverVersion.java

```diff
@@ -0,0 +1,52 @@
+package com.palantir.semver;
+
+public class SemverVersion {
+
+    private final String fullVersion;
+    private final String tagVersion;
+    private final String headCommitHash;
+    private final Integer count;
+    private final Integer buildNumber;
+    private final boolean isDirty;
+    private final boolean isVersionStableRelease;
+    
+    public SemverVersion (String fullVersion, String tagVersion, String headCommitHash,
+            Integer count, Integer buildNumber, boolean isDirty, boolean isVersionStableRelease) {
+        this.fullVersion = fullVersion;
+        this.tagVersion = tagVersion;
+        this.headCommitHash = headCommitHash;
+        this.count = count;
+        this.buildNumber = buildNumber;
+        this.isDirty = isDirty;
+        this.isVersionStableRelease = isVersionStableRelease;
+    }
+    
+    public String toString() {
+        return this.fullVersion;
```

Comment:
for the same reasons as the flex version object, i really with this was constructed from the parts rather than a field itself :(. I realize this is probably more complicated than flex versions so maybe we could punt this off but most of the logic should just be from TagBasedVersionFactory.generateVersion()